Tests/StreamChatTests/Controllers/CurrentUserController/CurrentUserController_Tests.swift

```diff
@@ -603,6 +603,64 @@ final class CurrentUserController_Tests: XCTestCase {
             XCTAssertEqual(reloadUserIfNeededCompletionError as? TestError, error)
         }
     }
+    
+    // MARK: - Mark all read
+    
+    func test_markAllRead_callsChannelListUpdater() {
+        
+        var completionCalled = false
+        weak var weakController = controller
+        
+        GIVEN("the completion has not been called yet") {
+            XCTAssertFalse(completionCalled)
+        }
+        
+        WHEN("all channels are marked as read succesfully and all potential references to the controller are gone") {
+            controller.markAllRead { [callbackQueueID] error in
+                AssertTestQueue(withId: callbackQueueID)
+                XCTAssertNil(error)
+                completionCalled = true
+            }
+                                    
+            controller = nil
+            
+            XCTAssertFalse(completionCalled)
+            
+            env.currentUserUpdater!.markAllRead_completion?(nil)
+            
+            env.currentUserUpdater!.markAllRead_completion = nil
+        }
+        
+        THEN("the completion is indeed called") {
+            AssertAsync.willBeTrue(completionCalled)
+        }
+        
+        AND("the reference to the controller is actually deallocated") {
+            AssertAsync.canBeReleased(&weakController)
+        }
```

Comment:
nit: would be nice to have a separate test for controller being kept alive until API response comes:
```swift
func test_loadPinnedMessages_keepsControllerAlive() {
        // Simulate `loadPinnedMessages` call
        controller.loadPinnedMessages { _ in }
        
        // Keep a weak ref so we can check if it's actually deallocated
        weak var weakController = controller
        
        // (Try to) deallocate the controller
        // by not keeping any references to it
        controller = nil
        
        // Assert controller is kept alive
        AssertAsync.staysTrue(weakController != nil)
    }
```